dev/io.openliberty.jakarta.xmlBinding.4.0/src/jakarta/xml/bind/ContextFinder.java

```diff
@@ -0,0 +1,550 @@
+            Object context = m.invoke(obj, classes, properties);
+            if (!(context instanceof JAXBContext)) {
+                // the cast would fail, so generate an exception with a nice message
+                throw handleClassCastException(context.getClass(), JAXBContext.class);
+            }
+            return (JAXBContext) context;
+
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new JAXBException(e);
+        } catch (InvocationTargetException e) {
+            // throw if it is exception not to be wrapped
+            // otherwise, wrap with a JAXBException
+            Throwable x = handleInvocationTargetException(e);
+
+            throw new JAXBException(x);
+        }
+    }
+
+    static JAXBContext find(String factoryId,
+                            String contextPath,
+                            ClassLoader classLoader,
+                            Map<String, ?> properties) throws JAXBException {
+
+        if (contextPath == null || contextPath.isEmpty()) {
+            // no context is specified
+            throw new JAXBException(Messages.format(Messages.NO_PACKAGE_IN_CONTEXTPATH));
+        }
+
+        //ModuleUtil is mr-jar class, scans context path for jaxb classes on jdk9 and higher
+        Class<?>[] contextPathClasses = ModuleUtil.getClassesFromContextPath(contextPath, classLoader);
+
+        String factoryName = classNameFromSystemProperties();
+        
+        // Liberty Change Start: Add Message log for warning if third-party impl is configured but not loaded
+        if (factoryName != null) {
+        	try {
+        		return newInstance(contextPath, contextPathClasses, factoryName, classLoader, properties);
+        	} catch (JAXBException x) {
+        		if(x.getMessage().contains(Messages.format(Messages.DEFAULT_PROVIDER_NOT_FOUND))) {
+              			// Only log the message about the system property being set but no implementation being found
+        			// as we want the RI to be used as a backup.
+                	Tr.warning(tc, JaxbToolsUtil.formatMessage(JaxbConstants.WARNING_SYSTEM_PROPERTY_JAXBCONTEXTFACTORY), factoryName, x.getMessage());
+        		} else {
+  			throw x;
+        		}  
+        	}
+        }
+
+        if (properties != null && !properties.isEmpty()) {
+            Object factory = properties.get(factoryId);
+            if (factory != null) {
+                if (factory instanceof String) {
+                    factoryName = (String) factory;
+                    
+                    if (factoryName != null) {        	
+                    	try {
+                    		return newInstance(contextPath, contextPathClasses, factoryName, classLoader, properties);
+                    	} catch (JAXBException x) {
+                			if(x.getMessage().contains(Messages.format(Messages.DEFAULT_PROVIDER_NOT_FOUND))) {
+                				// Only log the message about the system property being set but no implementation being found
```

Comment:
Does the spec say to do this?  It seems that the exception should be thrown and that is what the behavior should be if it isn't found.  Falling back to the RI seems wrong.